File: core/src/main/kotlin/io/holunda/connector/generic/GenericTask.kt
package io.holunda.connector.generic

import com.fasterxml.jackson.databind.*

data class GenericTask(
    val model: String,
    val context: JsonNode,
    val instructions: String,
    val output_schema: JsonNode,
) {
    companion object {
        fun fromRequest(request: GenericRequest) =
            GenericTask(
                request.model.modelId,
                request.inputJson,
                request.taskDescription,
                request.outputFormat
            )
    }
}

File: libs/bpm-ai-connectors-c8/bpm_ai_connectors_c8/tasks/foundational/generic/generic_task.py
from bpm_ai.generic.generic import run_generic
from bpm_ai_core.llm.common.llm import LLM
from pyzeebe import ZeebeTaskRouter

from bpm_ai_connectors_c8.config import DEFAULT_TASK_TIMEOUT

generic_router = ZeebeTaskRouter()


@generic_router.task(
    task_type="io.holunda:connector-generic:2",
    timeout_ms=DEFAULT_TASK_TIMEOUT
)
async def generic(
    model: LLM,
    inputJson: dict,
    taskDescription: str,
    outputFormat: dict
):
    return run_generic(
        llm=model,
        input_data=inputJson,
        instructions=taskDescription,
        output_schema=outputFormat
    )
